### PR TITLE
added specific tests for Events:: methods

### PR DESCRIPTION
**Pull request by dtacalau**
Signed-off-by: Daniel Tacalau <dst4096@gmail.com>

### CODE REVIEW

File: tests/events.rs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+#[test]
+fn events_all() {
+    let mut events = Events::with_capacity(16);
+    assert_eq!(events.capacity(), 16);
+    assert!(events.is_empty());
+
+    let mut poll = Poll::new().expect("unable to create new Poll instance");
+    let token = Token(10);
+    let waker = Waker::new(poll.registry(), token).expect("unable to create waker");
```
Comment (Thomasdezeeuw) [inline tests/events.rs]: Why not use `WAKE_TOKEN`?

File: tests/events.rs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+const WAKE_TOKEN: Token = Token(10);
+
+#[test]
+fn events_all() {
+    let mut events = Events::with_capacity(16);
+    assert_eq!(events.capacity(), 16);
+    assert!(events.is_empty());
+
+    let mut poll = Poll::new().expect("unable to create new Poll instance");
```
Comment (Thomasdezeeuw) [inline tests/events.rs]: Small nit; I don't think you need to use `expect`, `unwrap` will do.

File: tests/events.rs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        .unwrap();
+
+    assert!(!events.is_empty());
+
+    for event in events.iter() {
+        assert_eq!(event.token(), token);
+        assert!(event.is_readable());
+    }
+
+    assert!(!events.is_empty());
```
Comment (Thomasdezeeuw) [inline tests/events.rs]: This block tests the same thing as the for loop above.

File: tests/events.rs [hunk 4]
```diff
@@ -0,0 +1,35 @@
+use std::time::Duration;
+
+use mio::{Events, Poll, Token, Waker};
+
+const WAKE_TOKEN: Token = Token(10);
+
```
Comment (Thomasdezeeuw) [inline tests/events.rs]: Can you move the `event_source.rs` tests into this file? We're have too many tests file as it is and the file name is better.

File: tests/events.rs (hunk 1 above)
Comment (dtacalau) [reply tests/events.rs]: ok, done

File: tests/events.rs (hunk 2 above)
Comment (dtacalau) [reply tests/events.rs]: used init_with_poll.

File: tests/events.rs (hunk 3 above)
Comment (dtacalau) [reply tests/events.rs]: removed

File: tests/events.rs (hunk 4 above)
Comment (dtacalau) [reply tests/events.rs]: ok, done.

**Comment by dtacalau** [comment]
> LGTM. Blocked on #1192.

Rebased on master after #1192 merge.

**Pull request merged**